### PR TITLE
[Feature] Add right join

### PR DESCRIPTION
**Pull request by Morranto**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #7140 https://github.com/matrixorigin/MO-Cloud/issues/45

## What this PR does / why we need it:

### CODE REVIEW

**Review by fengttt — CHANGES_REQUESTED** [review]
This PR should not go in without extensive bvt tests.

**Comment by nnsgmsone** [comment]
> This PR should not go in without extensive bvt tests.

Right now plan doesn't support right join, so it may not be able to provide bvt @fengttt

File: pkg/common/hashmap/joinmap.go [hunk 1]
```diff
@@ -21,15 +21,16 @@ import (
...
+func NewJoinMap(sels [][]int64, nullSels []int64, expr *plan.Expr, mp *StrHashMap, hasNull bool, idx *index.LowCardinalityIndex) *JoinMap {
 	cnt := int64(1)
 	return &JoinMap{
-		cnt:     &cnt,
-		mp:      mp,
-		expr:    expr,
-		sels:    sels,
-		hasNull: hasNull,
-		idx:     idx,
+		cnt:      &cnt,
```
Comment (fengttt) [inline pkg/common/hashmap/joinmap.go:27]: why you need a &cnt?   A value is not good enough?

Comment (Morranto) [reply pkg/common/hashmap/joinmap.go:27]: this field is for reference count of joinmap which is shared by some scopes.   this field had been wrote by  @nnsgmsone   for about 60 days

**Review by ouyuanning — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Comment by Morranto** [comment]
> This PR should not go in without extensive bvt tests.

bvt cases have been added，please review this pr ，thank you

**Review by fengttt — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Pull request merged**